modules/01-filter.js

```diff
@@ -164,12 +164,15 @@ async function processCardAction(interaction) {
       await interaction.reply({ content: "Someone is already processing this flag!", ephemeral: true });
       return;
     }
-    processing.add(flag.id);
```

Comment:
This needs to stay the very first thing that happens (especially before any `await` calls) to prevent race conditions.